### PR TITLE
Added support for cancel-all command and its responses.

### PR DESCRIPTION
**Pull request by zarazek**

### CODE REVIEW

File: quedex_api/user_stream.py [hunk 1]
```diff
@@ -104,6 +104,22 @@ def on_order_cancel_failed(self, order_cancel_failed):
     """
     pass
 
+  def on_all_orders_cancelled(self, all_orders_cancelled):
+    """
+    :param all_orders_cancelled: dummy parameter (possibly empty dictionary)
```
Comment (wgromniak) [inline quedex_api/user_stream.py]: Misleading comment, I wouldn't know what to expect here. Make it an empty dict always and add explanation that this is for future extensibility?

File: quedex_api/user_stream.py [hunk 2]
```diff
@@ -234,6 +250,16 @@ def cancel_order(self, cancel_order_command):
     else:
       self._encrypt_send(cancel_order_command)
 
+  def cancel_all_orders(self):
+    if not self._initialized:
+      raise Exception('UserStream not initialized, wait until UserStreamListener.on_read is called')
+    cancel_all_orders_command = dict(type='cancel_all_orders')
```
Comment (wgromniak) [inline quedex_api/user_stream.py]: Not very pythonic way of creating a dict...

File: quedex_api/user_stream.py [hunk 3]
```diff
@@ -415,6 +443,8 @@ def check_modify_order(modify_order):
   if 'new_limit_price' not in modify_order and 'new_quantity' not in modify_order:
     raise ValueError('modify_order should have new_limit_price or new_quantity')
 
+def check_cancel_all_orders(cancel_all_orders):
+  return True
```
Comment (wgromniak) [inline quedex_api/user_stream.py]: Other `check_*` methods don't return anything, seams misleading and redundant...?

File: tests/test_user_stream.py [hunk 4]
```diff
@@ -96,6 +96,23 @@ def test_receiving_order_cancelled(self):
...
+  def test_receiving_all_orders_cancelled(self):
+    all_orders_cancelled = {'type': 'all_orders_cancelled'}
+    self.user_stream.on_message(self.serialize_to_trader([all_orders_cancelled]))
+
+    self.assertEqual(self.listener.error, None)
+    self.assertEqual(self.listener.all_orders_cancelled, all_orders_cancelled)
+    self.assertEqual(self.listener.message, all_orders_cancelled)
+
+  def test_receiving_cancel_all_orders_failed(self):
+    cancel_all_orders_failed = {'type': 'cancel_all_orders_failed',
```
Comment (wgromniak) [inline tests/test_user_stream.py]: Dict does not match the coding style in this file.

File: quedex_api/user_stream.py (hunk 1 above)
Comment (zarazek) [reply quedex_api/user_stream.py]: Expect the unexpected! Personally I don't want this parameter here, but it is required by generic listener invoking mechanism. We just pass whole event dictionary here, which in this case contains is only {"type": "all_orders_cancelled"}. But there is nothing said about "type" key in other comments. So maybe i would just say "dummy parameter, reserved for future extension" ?

File: quedex_api/user_stream.py (hunk 2 above)
Comment (zarazek) [reply quedex_api/user_stream.py]: And the pythonic is { key: value } ?

File: tests/test_user_stream.py (hunk 4 above)
Comment (zarazek) [reply tests/test_user_stream.py]: What's different? It's just like in the previous tests (i. e. test_receiving_order_cancelled), it just doesn't fit in one line.

File: quedex_api/user_stream.py (hunk 3 above)
Comment (zarazek) [reply quedex_api/user_stream.py]: OK, I'll change it to 'pass'

File: quedex_api/user_stream.py (hunk 2 above)
Comment (wgromniak) [reply quedex_api/user_stream.py]: Yes.

File: tests/test_user_stream.py (hunk 4 above)
Comment (wgromniak) [reply tests/test_user_stream.py]: I wasn't clear, this way:
```
cancel_all_orders_failed = {
  'type': 'cancel_all_orders_failed',
  'cause': 'session_not_active',
}
```

Comment (zarazek) [reply tests/test_user_stream.py]: OK

**Comment by zarazek** [comment]
Adressed all the comments

File: quedex_api/user_stream.py [hunk 5]
```diff
@@ -234,6 +250,16 @@ def cancel_order(self, cancel_order_command):
     else:
       self._encrypt_send(cancel_order_command)
 
+  def cancel_all_orders(self):
+    if not self._initialized:
+      raise Exception('UserStream not initialized, wait until UserStreamListener.on_read is called')
```
Comment (wgromniak) [inline quedex_api/user_stream.py]: Typo: `on_ready`.

Comment (zarazek) [reply quedex_api/user_stream.py]: Not only here... Fixing...

File: quedex_api/user_stream.py [hunk 6]
```diff
@@ -416,6 +411,9 @@ def _call_listeners(self, method_name, *args, **kwargs):
       if hasattr(listener, method_name):
         getattr(listener, method_name)(*args, **kwargs)
 
+  def _check_if_initialized(self):
```
Comment (wgromniak) [inline quedex_api/user_stream.py]: +1

**Pull request merged**